### ISSUE TITLE
An adjustable bar's thumb is cut-off on the left if inside a viewport

### ISSUE THREAD

**Issue by Tayruu** [renpy/renpy#5735]
I was boggling as to why my options' screen **thumb** image was cutting off out of the blue, until I eventually found that it was specifically happening inside a **viewport**. This is assuming a **thumb_offset**  value exists so that the thumb can overlap the ends of the slider.

Here is a demo.
```python
label start:
  
    show screen computer
    
    $ ui.interact()
    
    "Closing."

    return   

style slider:
    thumb           Flatten("#aaa", xysize=(32, 32))
    thumb_offset    16

screen computer:
    window:
        pos(0.5, 0.5) xpadding 128
        
        has vbox
        
        label "Standard Frame"
        
        frame:
            xfill True
            has side "l r c"
            spacing 16
            frame:
                xsize 200
                text "Music Volume"
            frame:
                xsize 100
                text ("%d" % (_preferences.get_volume('music') * 100)) xalign 1.0
            frame:
                xfill True
                bar value Preference("music volume")

        label "Viewport Behaviour"

        viewport:
            frame:
                xfill True
                has side "l r c"
                spacing 16
                frame:
                    xsize 200
                    text "Music Volume"
                frame:
                    xsize 100
                    text ("%d" % (_preferences.get_volume('music') * 100)) xalign 1.0
                frame:
                    xfill True
                    bar value Preference("music volume")
```
![screenshot0001](https://github.com/user-attachments/assets/061fa7ea-ca36-4cae-a95e-5f3c3058eba8)

**Comment by mal** [comment]
The `thumb_offset` value is the number of pixels the bars (left or right) should extend beneath the thumb, where a value half the width of the thumb would cause them to touch. It's used to prevent the bars ending at the edge of none-rectangular thumbs, or thumbs that include transparency.

The thumb being clipped inside the viewport is expected, as that's what viewports do, clip their contents. The value you instead want to alter is gutter, which can be used to stop a thumb before it exceeds the bounds of the bar (i.e. before it would be clipped by the viewport).

**Comment by renpytom** [comment]
I'm convinced this is a real error.

**Comment by renpytom** [comment]
Fixed by a53b068.
